Clarify docs regarding non-CBTF PrairieTest sessions (#10070)
PrairieLearn/PrairieLearn: docs/accessControl.md

Each user accesses the PrairieLearn server in a `mode`, as listed below. This can be used to restrict access to assessments based on the current mode. The `mode` setting of an access rule has the following effect:

| Mode        | When is access allowed?                                                                                                                                                                                                                                                               |
| ----------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| `Public`    | Access is normally allowed, but is blocked when the student is checked in on PrairieTest, either in a Computer-Based Testing Facility (CBTF) or in a course-managed session.                                                                                                          |

> | `Exam`      | Access is normally blocked, and is only allowed when the student is checked in on PrairieTest, either in a Computer-Based Testing Facility (CBTF) on in a course-managed session. The `examUuid` should also be specified in the same access rule to limit access to a specific exam. |

Reviewer: In PT we always use the phrase "testing center" so I think it might be better to never use the phrase CBTF and instead write something like:
```suggestion
| `Exam`      | Access is normally blocked, and is only allowed when the student is checked in on PrairieTest, either in a testing center or in a course-managed session. The `examUuid` should also be specified in the same access rule to limit access to a specific exam. |
```

Author: Done in a5aedf4c8 in multiple locations. I had to be careful not to say "centre", let me know if you see a slip... :grin:

> | no mode set | Access is allowed anytime, both during PrairieTest exams and outside of exams.                                                                                                                                                                                                        |

Reviewer 2: This is no longer true as of #9743. If `"mode"` is not set in an access rule, it will never be visible when a student is in Exam mode because of PrairieTest.

The subtlety here is that if a student is in Exam mode because they're accessing via a legacy `exam_mode_networks` IP, the old behavior takes effect (a rule without an explicit `mode` will be accessible). I don't know if it's worth putting this in the documentation though, this is virtually never used anymore.

@mwest1066 did I get this all correct? Do you have any better thoughts on how to document this for users?

Author: So other than networks, what's the difference between no mode and public mode?

Reviewer: Ignoring legacy environments, there is no longer a difference between no mode and public mode. That is, if we have an `exam_uuid` set then we must have `mode = Exam` (it's an error to have anything else). If we don't have an `exam_uuid` then we must have no mode or `mode = Public` and both are treated exactly the same.

@nwalters512 I wonder whether we should change the code (no necessarily immediately) to treat "no mode + `exam_uuid`" as valid and equivalent to the current "`mode = Exam` + `exam_uuid`". This would allow us to remove the documentation about `mode` completely from access rules. We would still retain the code that uses `mode` for legacy environments.

Reviewer 2: @mwest1066 I've been wondering about that as well; it'd be a nice conceptual simplification!

Author: TBH, that sounds like a great idea. If for nothing else, it would remove the confusion between `"mode": "Exam"` and `"type": "Exam"` until #3347 is addressed.

Author: Docs updated in a5aedf4c8.

> In general usage it is best to set `"mode": "Public"` for any homework (assessments that students should do at home or without special access control), and to set both `"mode": "Exam"` and `"examUuid"` for exams with access controlled by PrairieTest. This will make it so that exams are only accessible in the CBTF or a controlled course session, and homework is _not_ accessible during exams.

Reviewer 2: ```suggestion
In general usage it is best to set `"mode": "Public"` for any homework (assessments that students should do at home or without special access control), and to set both `"mode": "Exam"` and `"examUuid"` for exams with access controlled by PrairieTest. This will make it so that exams are only accessible in a CBTF or a controlled course session, and homework is _not_ accessible during exams.
```

> In general usage it is best to set `"mode": "Public"` for any homework (assessments that students should do at home or without special access control), and to set both `"mode": "Exam"` and `"examUuid"` for exams with controlled accessed by PrairieTest. This will make it so that exams are only accessible in the CBTF or a controlled course session, and homework is _not_ accessible during exams.

Author: ```suggestion
In general usage it is best to set `"mode": "Public"` for any homework (assessments that students should do at home or without special access control), and to set both `"mode": "Exam"` and `"examUuid"` for exams with access controlled by PrairieTest. This will make it so that exams are only accessible in the CBTF or a controlled course session, and homework is _not_ accessible during exams.
```

![Time limit illustrations](exam_timer.svg)

> **Note that time limits should not be set for exams in the CBTF (Computer-Based Testing Facility) or course-managed PrairieTest sessions. Instead, such exams should set `"mode": "Exam"`, in which case `timeLimitMin` will have no effect and the time limits will be enforced by PrairieTest.**

Reviewer 2: ```suggestion
**Note that time limits should not be set for exams in a CBTF (Computer-Based Testing Facility) or course-managed PrairieTest sessions. Instead, such exams should set `"mode": "Exam"`, in which case `timeLimitMin` will have no effect and the time limits will be enforced by PrairieTest.**
```

## Exam UUIDs

> To require that students take a particular exam in the Computer-Based Testing Facility (CBTF) or a course-managed PrairieTest session, the `examUuid` should be set to the UUID value from PrairieTest. For example:

Reviewer 2: ```suggestion
To require that students take a particular exam in a Computer-Based Testing Facility (CBTF) or a course-managed PrairieTest session, the `examUuid` should be set to the UUID value from PrairieTest. For example:
```

],
```

> The above `allowAccess` directive is appropriate for an `Exam` being taken by on-campus students in the CBTF (or a course-managed PrairieTest session) and by remote students. First, anyone (i.e., on-campus students) can access the assessment in the CBTF (`"mode": "Exam"`) for full credit. Second, a defined set of students (remote students) can take the exam for full credit on Sept 10th. For the off-campus students we set a time limit (50 minutes). For on-campus students no time limit should be given because the time limit is enforced by PrairieTest.

Reviewer 2: ```suggestion
The above `allowAccess` directive is appropriate for an `Exam` being taken by on-campus students in a CBTF (or a course-managed PrairieTest session) and by remote students. First, anyone (i.e., on-campus students) can access the assessment in a CBTF (`"mode": "Exam"`) for full credit. Second, a defined set of students (remote students) can take the exam for full credit on Sept 10th. For the off-campus students we set a time limit (50 minutes). For on-campus students no time limit should be given because the time limit is enforced by PrairieTest.
```
